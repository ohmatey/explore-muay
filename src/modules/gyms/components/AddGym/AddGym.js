import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-final-form'

import AddGymForm from '../AddGymForm'

const AddGym = ({
  handleFormSubmit
}) => (
  <div>
    <Form
      onSubmit={handleFormSubmit}
      component={AddGymForm}
    />
  </div>
)

AddGym.propsTypes = {
  handleFormSubmit: PropTypes.func.isRequired
}

export default AddGym
