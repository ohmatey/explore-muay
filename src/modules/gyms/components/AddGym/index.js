import AddGym from './AddGym'

export default AddGym
