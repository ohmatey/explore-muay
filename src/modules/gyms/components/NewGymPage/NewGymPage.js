import React from 'react'

import AddGym from '../AddGym'

const NewGymPage = ({
  handleFormSubmit
}) => (
  <div>
    <AddGym handleFormSubmit={handleFormSubmit} />
  </div>
)

export default NewGymPage
