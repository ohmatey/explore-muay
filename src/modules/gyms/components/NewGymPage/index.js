import NewGymPage from './NewGymPage'

export default NewGymPage
