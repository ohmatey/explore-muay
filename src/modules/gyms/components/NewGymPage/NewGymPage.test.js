import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import NewGymPage from './NewGymPage'
import AddGym from '../AddGym'

describe('<NewGymPage />', () => {
  let component

  beforeEach(() => {
    component = shallow(<NewGymPage />)
  })

  it('should render with no errors', () => {
    const tree = toJson(component)

    expect(tree).toMatchSnapshot()
  })

  it('should render with add gym form', () => {
    expect(component.find(AddGym).length).toEqual(1)

    const mockSubmit = jest.fn()

    component.setProps({
      handleFormSubmit: mockSubmit
    })

    expect(component.find(AddGym).prop('handleFormSubmit')).toEqual(mockSubmit)
  })
})
