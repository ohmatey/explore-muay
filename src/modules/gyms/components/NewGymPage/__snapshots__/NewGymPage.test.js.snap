// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGymPage /> should render with no errors 1`] = `
<div>
  <AddGym />
</div>
`;
